spec/controllers/sessions_controller_spec.rb

```diff
@@ -161,88 +195,132 @@
+            expect(user.applicant?).to be_truthy
 
-          expect(user.username).to be_present
-          expect(authentication.provider).to be_present
-          expect(authentication.uid).to be_present
-        end
+            expect(user.username).to eq("someone")
+            expect(user.email).to eq("someone@foo.bar")
+            expect(user.email_for_google).to eq(nil)
+            expect(authentication.provider).to eq("github")
+            expect(authentication.uid).to eq("12345")
+          end
 
-        it "sets the session with the newly-created user" do
-          subject
-          expect(session[:user_id]).to eq(user.id)
+          it "sets the session with the newly-created user" do
+            subject
+            expect(session[:user_id]).to eq(user.id)
+          end
         end
-      end
 
-      context "with an existing user" do
-        let(:user) { create_with_omniauth(OmniAuth.config.mock_auth[:github]) }
+        context "with an existing user" do
+          let(:user) { create_with_omniauth(OmniAuth.config.mock_auth[:github]) }
 
-        before do
-          user.update_attribute(:state, "member")
+          before do
+            user.update_attribute(:state, "member")
+          end
+
+          it "does not create a user" do
+            expect { subject }.not_to change { User.count }
+          end
+
+          it "does not set the session" do
+            subject
+            expect(session[:user_id]).to be_nil
+          end
+
+          it "redirects to the root path" do
+            expect(subject).to redirect_to :root
+          end
+
+          it "sets the flash message" do
+            subject
+            expect(flash[:alert]).to include "It looks like you've previously logged in"
+          end
         end
+      end
 
-        it "does not create a user" do
-          expect { subject }.not_to change { User.count }
+      context "with a bad email address" do
+        let(:email) { "someone" }
+
+        it "rerenders the page" do
+          expect(subject).to render_template :get_email
         end
 
         it "does not set the session" do
           subject
           expect(session[:user_id]).to be_nil
         end
+      end
 
-        it "redirects to the root path" do
-          expect(subject).to redirect_to :root
+      context "with no email address" do
+        let(:email) { "" }
+
+        it "rerenders the page" do
+          expect(subject).to render_template :get_email
         end
 
-        it "sets the flash message" do
+        it "does not set the session" do
           subject
-          expect(flash[:alert]).to include "It looks like you've previously logged in"
+          expect(session[:user_id]).to be_nil
         end
       end
     end
 
-    context "with a bad email address" do
-      let(:email) { "someone" }
-
-      it "rerenders the page" do
-        expect(subject).to render_template :get_email
+    describe "with Google authentication" do
```

Comment:
And similar to above -- this describe "with Google authentication" block (lines 290-324) is completely new -- it's confusing that Github diffs it against other test cases that were already there.